apps/mobile/src/components/Camera/QrCamera.tsx

```diff
@@ -0,0 +1,202 @@
+    await Linking.openSettings()
+  }, [])
+
+  const denied = permission === 'denied'
+
+  return (
+    <Theme name={'dark'}>
+      <View style={styles.container}>
+        {/* Camera View */}
+        {device && (
+          <Camera style={StyleSheet.absoluteFill} device={device} isActive={isCameraActive} codeScanner={codeScanner} />
+        )}
+
+        {/* Overlay with blurred edges */}
+        <View style={styles.overlay}>
+          <View flex={1}>
+            <BlurView
+              style={[styles.blurTop, denied && { backgroundColor: 'rgba(0, 0, 0, 1)' }]}
+              intensity={30}
+              // experimentalBlurMethod={'dimezisBlurView'}
+              tint={'systemUltraThinMaterialDark'}
+            >
+              <View style={styles.topContainer}>
+                <View style={{ flex: 1, marginTop: 30, marginLeft: 20, flexDirection: 'row' }}>
+                  <Pressable
+                    onPress={() => {
+                      router.back()
+                    }}
+                  >
+                    <Badge
+                      themeName="badge_background"
+                      circleSize="$9"
+                      content={<SafeFontIcon size={20} name="close" />}
+                    />
+                  </Pressable>
+                </View>
+
+                <View flex={1} justifyContent={'flex-end'} alignItems={'center'} marginBottom={'$8'}>
+                  {typeof heading === 'string' ? <H3>{heading}</H3> : heading}
+                </View>
+              </View>
+            </BlurView>
+
+            {/* Middle with transparent center */}
+            <View style={styles.transparentCenter}>
+              <BlurView
+                style={[styles.sideBlur, denied && { backgroundColor: 'rgba(0, 0, 0, 1)' }]}
+                intensity={30}
+                // experimentalBlurMethod={'dimezisBlurView'}
+                tint={'systemUltraThinMaterialDark'}
+              />
+
+              <View style={[styles.transparentBox, denied && { backgroundColor: 'rgba(0, 0, 0, 0.8)' }]}>
+                {/* Green corners */}
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.topLeft]} />
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.topRight]} />
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.bottomLeft]} />
+                <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.bottomRight]} />
+
+                {denied && (
+                  <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
+                    <SafeFontIcon name={'camera'} size={40} color={'$error'} />
+                    <SafeButton rounded secondary onPress={openSettings} marginTop={20}>
+                      Enable camera
+                    </SafeButton>
+                  </View>
+                )}
+              </View>
```

Comment:
Shall we move it to a component called NoCameraPermission or something like that?

Reply:
I went for CameraLens as this is the area that has no blur when permissions are on.